cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactCodec.java

```diff
@@ -118,16 +125,39 @@ public FactCodec(EntityTable entityTable, int resolution, int rollTimebaseInterv
     return rowKey;
   }
 
+  public byte[] getNextRowKey(List<TagValue> previousTags, byte[] rowKey) {
+    byte[] newRowKey = new byte[rowKey.length];
+    int offset = entityTable.getIdSize() + Bytes.SIZEOF_INT +  entityTable.getIdSize() * previousTags.size();
+    byte[] stopKey = Bytes.stopKeyForPrefix(Arrays.copyOfRange(rowKey,
```

Comment:
yeah, the assumption is fine, as we have the tags matching, and the next field can we either a tag or metric, we can continue incrementing and searching, but we add to the returned list only tags.